quinn/src/notify.rs

```diff
@@ -0,0 +1,89 @@
+use std::sync::{Arc, Mutex};
+use std::task::{Context, Waker};
+
+use slab::Slab;
```

Comment:
A slab is required so that `Waiter`s can reliably and efficiently update and remove their associated `Waker`s on poll and drop respectively without invalidating eachothers' indices. If wakers can't be updated, then we'll miss wakeups after e.g. tokio workstealing, and if they can't be removed, then memory use grows without bound in some use patterns; consider e.g. a `loop { select! { ... } }` which constructs new `Waiter`s on each cycle but usually selects a different branch.

Reply:
Mooted by adoption of `tokio::sync::Notify`.